org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/breakpoints/ScalaDebugBreakpointManager.scala

```diff
@@ -88,6 +96,8 @@ private[debug] object ScalaDebugBreakpointManagerActor {
   case class BreakpointAdded(breakpoint: IBreakpoint)
   case class BreakpointRemoved(breakpoint: IBreakpoint)
   case class BreakpointChanged(breakpoint: IBreakpoint, delta: IMarkerDelta)
+  case class ReenableBreakpointsAfterHcr(classNames: Seq[String])
+  case object ReenableBreakpointAfterHcr
```

Comment:
The class is used when sending a message to `ScalaDebugBreakpointManagerActor` and the object is used when sending a message to `BreakpointSupportActor` from `ScalaDebugBreakpointManagerActor` (in this case we don't need to add an additional information).

In other words, it's not that the object is used, when classNames is empty. When it's empty, we don't need to perform HCR. Hence, we also don't need to send these messages.

Do you prefer to describe them or maybe simply the object should be moved to `BreakpointSupportActor` object? WDYT?